blender/LilySurfaceScrapper/Scrappers/Cc0texturesScrapper.py

```diff
@@ -34,51 +36,44 @@ def canHandleUrl(cls, url):
     def fetchVariantList(self, url):
         """Get a list of available variants.
         The list may be empty, and must be None in case of error."""
-        html = self.fetchHtml(url)
-        if html is None:
+        if Cc0texturesScrapper._texture_cache is None:
+            Cc0texturesScrapper._texture_cache = self.fetchXml("https://cc0textures.com/api/getDownloadLinks.php")
+
+        root = Cc0texturesScrapper._texture_cache
+
+        texture = root.find(f"assets/item[link='{url}']")
+
+        if texture is None:
             return None
-        
-        # Get variants
-        variants_data = []
-        variants = []
-        for button in html.xpath("//h3[@class='downloads']//a"):
-            name = button.xpath("text()")[0].strip()
-            if name == 'SBSAR':
-                continue
-            variants_data.append(button)
-            variants.append(name)
-        
-        # Save some data for fetchVariant
-        self._html = html
-        self._variants_data = variants_data
-        return variants
+
+        self._texture = texture
+        self._variants = list(filter(lambda v: v.attrib["res"] in ['1', '2', '4', '8'], texture.findall("downloads-pretty/download")))
+        return list(map(lambda v: v.attrib["res"], self._variants))
     
     def fetchVariant(self, variant_index, material_data):
         """Fill material_data with data from the selected variant.
         Must fill material_data.name and material_data.maps.
         Return a boolean status, and fill self.error to add error messages."""
         # Get data saved in fetchVariantList
-        html = self._html
-        variants_data = self._variants_data
-        
-        if variant_index < 0 or variant_index >= len(variants_data):
+        texture = self._texture
+        variants = self._variants
+
+        if variant_index < 0 or variant_index >= len(variants):
             self.error = "Invalid variant index: {}".format(variant_index)
             return False
-        v = variants_data[variant_index]
-        base_name = html.xpath("//div[@class='information']/h1/text()")[0].replace('#', '')
-        variant_name = v.xpath("text()")[0].strip().replace('|', '')
         
-        material_data.name = "CC0Textures/" + base_name + "/" + variant_name
-        
-        zip_url = "https://cc0textures.com" + v.attrib['href'][1:]
+        variant = variants[variant_index]
+
+        material_data.name = "CC0Textures/" + texture.find("title").text + "/" + variant.attrib["res"]
```

Comment:
Might be needed to escape some caracters here like done previously for sharps.
```suggestion
        base_name = texture.find("title").text.replace('#', '')
        material_data.name = "CC0Textures/" + base_name + "/" + variant.attrib["res"]
```